includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php

```diff
@@ -1366,10 +1366,21 @@ public function add_contact( $contact, $list_id = false, $sublists = [] ) {
 		}
 
 		$new_contact_status = 'subscribed';
-		if ( isset( $contact['metadata'] ) && ! empty( $contact['metadata']['status'] ) ) {
+		$status_if_new_informed = false;
```

Comment:
I find it confusing to add another variable to the mix. If `status_if_new` is provided, we can just add to `$update_payload`, same with `status`. I don't think we need `$new_contact_status` at all.

When populating those items, the developer should be as worried as they would be when actually making the Mailchimp request.

Reply:
yes, that was because we started handling statuses before handling metadata, but then would only add to the payload later, because the payload variable didn't exist at this point.

I changed that and it looks simpler now